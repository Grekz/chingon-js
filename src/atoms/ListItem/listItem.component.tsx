import styled from '@emotion/styled'
import * as React from 'react'

interface IListItemProps {
  activeBackgroundColor?: string
  activeBorderColor?: string
  activeColor?: string
  backgroundColor?: string
  borderColor?: string
  children: React.ReactNode
  color?: string
  hoverBackgroundColor?: string
  isActive?: boolean
}

export type ListItemType = React.SFC<IListItemProps>

const ListItem = (props: IListItemProps) => (
  <ListItemWrapper {...props}>{props.children}</ListItemWrapper>
)

ListItem.defaultProps = {
  activeBackgroundColor: '#007bff',
  activeBorderColor: '#007bff',
  activeColor: '#fff',
  backgroundColor: '#fff',
  borderColor: 'rgba(0, 0, 0, 0.125)',
  color: '#000',
  hoverBackgroundColor: '#f8f9fa',
  isActive: false,
}

const ListItemWrapper: ListItemType = styled.div`
  position: relative;
  display: block;
  padding: 0.75rem 1.25rem;
  margin-bottom: -1px;
  background-color: ${props =>
    props.isActive ? props.activeBackgroundColor : props.backgroundColor};
  border: 1px solid
    ${props => (props.isActive ? props.activeBorderColor : props.borderColor)};
  color: ${props => (props.isActive ? props.activeColor : props.color)};
  &:first-child {
    border-top-left-radius: 0.25rem;
    border-top-right-radius: 0.25rem;
  }
  &:last-child {
    border-bottom-right-radius: 0.25rem;
    border-bottom-left-radius: 0.25rem;
  }
  &:hover,
  &:focus {
    z-index: 1;
    text-decoration: none;
    background-color: ${props =>
      props.isActive
        ? props.activeBackgroundColor
        : props.hoverBackgroundColor};
    color: ${props => (props.isActive ? props.activeColor : props.color)};
  }
`

export default ListItem
