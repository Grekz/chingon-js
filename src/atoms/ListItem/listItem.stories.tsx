import * as React from 'react'

// tslint:disable-next-line: no-implicit-dependencies
import { storiesOf } from '@storybook/react'

import ListItem from '.'

storiesOf('Atoms|ListItem', module).add('As Default', () => (
  <>
    <ListItem>List item</ListItem>
    <ListItem isActive={true}>Active list item</ListItem>
    <ListItem>List item</ListItem>
  </>
))
