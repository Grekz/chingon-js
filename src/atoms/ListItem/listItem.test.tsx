import { configure, shallow } from 'enzyme'
import * as Adapter from 'enzyme-adapter-react-16'
import toJson from 'enzyme-to-json'
import serializer, { matchers } from 'jest-emotion'
import * as React from 'react'
import ListItem from '.'

configure({ adapter: new Adapter() })
expect.extend(matchers)
expect.addSnapshotSerializer(serializer)

describe('<ListItem /> rendering', () => {
  it('should render ListItem', () => {
    const wrapper = shallow(<ListItem>List item</ListItem>)
    expect(wrapper).toHaveLength(1)
  })
  it('should render active ListItem', () => {
    const wrapper = shallow(<ListItem isActive>List item</ListItem>)
    expect(wrapper).toHaveLength(1)
  })
  it('should match ListItem snapshot', () => {
    const wrapper = shallow(<ListItem>List item</ListItem>)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
