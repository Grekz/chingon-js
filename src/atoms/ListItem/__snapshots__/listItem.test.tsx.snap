// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListItem /> rendering should match ListItem snapshot 1`] = `
<Styled(div)
  activeBackgroundColor="#007bff"
  activeBorderColor="#007bff"
  activeColor="#fff"
  backgroundColor="#fff"
  borderColor="rgba(0, 0, 0, 0.125)"
  color="#000"
  hoverBackgroundColor="#f8f9fa"
  isActive={false}
>
  List item
</Styled(div)>
`;
